experiments/functional/rothreshold-limit/test.yml

```diff
@@ -0,0 +1,432 @@
+              args:
+                executable: /bin/bash               
+              register: app_mount_path
+
+            - name: Fetch the Storage from PVC using namespace
+              shell: kubectl get pvc -n {{ app_ns }} -o jsonpath={.items[0].spec.resources.requests.storage}
+              args:
+                executable: /bin/bash
+              register: storage_capacity
+
+            - name: Fetch the alphabet(G,M,m,g) from storage capacity
+              shell: echo "{{ storage_capacity.stdout }}" | grep -o -E '[0-9]+'
+              args:
+                executable: /bin/bash
+              register: value_str
+
+            - set_fact:
+                value_num: '{{ (((value_str.stdout | int ) * 1024) * 1024) |  int }}'
+                 
+            - name: Writing data on application mount point
+              shell: >
+                kubectl exec -it {{ app_pod_name.stdout }} -n {{ app_ns }} 
+                -- sh -c "cd {{ app_mount_path.stdout }} && dd if=/dev/urandom of=test.txt bs=4k count={{ value_num }}"
+              args:
+                executable: /bin/bash
+
+            - name: Check the volume is successfully filled
+              shell: kubectl logs {{ app_pod_name.stdout }} -n {{ app_ns }} | grep 'No space left on device'
+              args:
+                executable: /bin/bash
+              register: space
+              until: "'No space left on device' in space.stdout"
+              retries: 150
+              delay: 5
+
+            - name: Verify the readOnly status of CSP after reached the threshold limit
+              shell: >
+                kubectl get csp -o jsonpath='{.items[?(@.metadata.name=="{{ item }}")].status.readOnly}'
+              args:
+                executable: /bin/bash
+              register: aft_csp_status
+              with_items: "{{ pool_deployment.stdout_lines }}"
+              until: "'true' in aft_csp_status.stdout"
+              delay: 5
+              retries: 30
+
+            - name: Patch the csp to change the threshold value
+              shell: >
+                kubectl patch csp {{ item }} --type='json' -p='[{"op":"replace", "path":"/spec/poolSpec/roThresholdLimit", "value":85}]'
+              args:
+                executable: /bin/bash
+              register: patch_csp_status
+              with_items: "{{ pool_deployment.stdout_lines }}"
+              failed_when: "'patched' not in patch_csp_status.stdout"
+
+            - name: Verify the readOnly status of CSP after changed it to default
+              shell: >
+                kubectl get csp -o jsonpath='{.items[?(@.metadata.name=="{{ item }}")].status.readOnly}'
+              args:
+                executable: /bin/bash
+              register: csp_status
+              with_items: "{{ pool_deployment.stdout_lines }}"
+              until: "'false' in csp_status.stdout"
+              delay: 5
+              retries: 30              
+                                        
```

Comment:
Can we delete the application pod and check if the volume can be mounted again successfully